AirCasting/SessionViews/ABMeasurementsView.swift

```diff
@@ -3,27 +3,51 @@
 
 import SwiftUI
 import AirCastingStyling
-import CoreLocation
-import CoreData
 
 enum MeasurementPresentationStyle {
     case showValues
     case hideValues
 }
 
 struct ABMeasurementsView: View {
+    var viewModelProvider: () -> CardMeasurementsViewModel
     @ObservedObject var session: SessionEntity
     @Binding var isCollapsed: Bool
     @Binding var selectedStream: MeasurementStreamEntity?
-    @Binding var showLoadingIndicator: Bool
     var thresholds: [SensorThreshold]
     let measurementPresentationStyle: MeasurementPresentationStyle
-    let sessionDownloader = SessionDownloadService(client: URLSession.shared,
-                                                   authorization: UserAuthenticationSession(),
-                                                   responseValidator: DefaultHTTPResponseValidator())
-    let measurementStreamStorage: MeasurementStreamStorage
-    @EnvironmentObject var selectedSection: SelectSection
 
+    @State private var viewModel: CardMeasurementsViewModel?
+    
+    var body: some View {
+        ZStack {
+            if let viewModel = viewModel {
+                _ABMeasurementsView(measurementsViewModel: viewModel,
+                                    session: session,
+                                    isCollapsed: $isCollapsed,
+                                    selectedStream: $selectedStream,
+                                    thresholds: thresholds,
+                                    measurementPresentationStyle: measurementPresentationStyle)
+            }
+        }
+        .onAppear {
+            viewModel = viewModelProvider()
+        }
+    }
+}
+
+struct _ABMeasurementsView: View {
```

Comment:
not sure why the split into `_ABMeasurementsView` and `ABMeasurementsView`  🤔

Reply:
It's used for view model provider, to avoid the situation described in the comment above ^